src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs

```diff
@@ -109,8 +110,19 @@ public void Start<TContext>(IHttpApplication<TContext> application)
 
                     if (!parsedAddress.Host.Equals("localhost", StringComparison.OrdinalIgnoreCase))
                     {
-                        _disposables.Push(engine.CreateServer(
-                            parsedAddress));
+                        try
+                        {
+                            _disposables.Push(engine.CreateServer(parsedAddress));
+                        }
+                        catch (AggregateException ex)
+                        {
+                            if ((ex.InnerException as UvException)?.StatusCode == Constants.EADDRINUSE)
+                            {
+                                throw new IOException($"Failed to bind to address {parsedAddress}: address already in use.", ex);
```

Comment:
Startup is weird, there's really no standard exception you can throw. IO isn't terrible, at least it's related to the socket.